src/edu/csus/ecs/pc2/clics/API202306/ProblemService.java

```diff
@@ -0,0 +1,143 @@
+        // public only gets the problems when the contest starts
+        if (sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) || sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE) || model.getContestTime().isContestStarted()) {
+            for (Problem problem: problems) {
+                if (problem.isActive()) {
+                    problist.add(new CLICSProblem(model, problem, ord));
+                    ord++;
+                }
+            }
+        }
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            String json = mapper.writeValueAsString(problist);
+            return Response.ok(json, MediaType.APPLICATION_JSON).build();
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for problems " + e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Return the reponse to a request for a single problem's information for the specified contest and problem id.
+     *
+     * @param sc user info
+     * @param contestId the contest
+     * @param problemId the problem id desired
+     * @return response
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{problemId}/")
+    public Response getProblem(@Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("problemId") String problemId) {
+        // public only gets the problems when the contest starts
+        if (sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) || sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE) || model.getContestTime().isContestStarted()) {
+            // get the problems from the contest
+            Problem[] problems = model.getProblems();
+            int ord = 1;
+            for (Problem problem: problems) {
+                // match by ID
+                if (problem.isActive()) {
+                    if(IJSONTool.getProblemId(problem).equals(problemId)) {
+                        try {
+                            return Response.ok(JSONUtilities.getObjectMapper().writeValueAsString(new CLICSProblem(model, problem, ord)), MediaType.APPLICATION_JSON).build();
+                        } catch(Exception e) {
+                            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for problem " + problemId + " in contest " + contestId + ": " + e.getMessage()).build();
+                        }
+                    }
+                    ord++;
+                }
+            }
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        return Response.status(Response.Status.FORBIDDEN).build();
+    }
+
+    /**
+     * Retrieve access information about this endpoint for the supplied user's security context
+     *
+     * @param sc User's security information
+     * @return CLICSEndpoint object if the user can access this endpoint's properties, null otherwise
+     */
+    public static CLICSEndpoint getEndpointProperties(SecurityContext sc) {
```

Comment:
Same comment as in other files:  `SecurityContext` is not being used (but the method header comments imply it is).  Or else I don't understand what's going on...

Reply:
Same reply as other files.